### DISCUSSION TITLE
Annoying logs

### DISCUSSION THREAD

**Discussion by iposokhin** [parcel-bundler/parcel#7804] [Q&A]
I added `--log-level verbose` flag and now see warnings related with lib dependencies, like React, ReactDOM and so on. Can I disable logs for libraries? Anyway I can't fix them. Can I see logs related only with my own codebase

```
// node_modules/react/index.js

if (process.env.NODE_ENV === 'production') {
  module.exports = require('./cjs/react.production.min.js');
} else {
  module.exports = require('./cjs/react.development.js');
}
```

```
// logs

@parcel/transformer-js: Conditional or non-top-level `require()` call. This causes the resolved module and all dependendencies to be wrapped.

  /home/ipos/issue-reproduce/node_modules/react/index.js:6:20
    5 | } else {
  > 6 |   module.exports = require('./cjs/react.development.js');
  >   |                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    7 | }
    8 | 

  📝 Learn more: https://parceljs.org/features/scope-hoisting/#avoid-conditional-require()
```

**Comment by devongovett** [comment]
Turn off log-level verbose? 😉

**Reply by iposokhin** [reply]
> Turn off log-level verbose? wink

It's obviously solution :smile: , but I want see logs related with my codebase to fix them and avoid issues with tree shaking

**Reply by muhammadnasr** [reply]
I have very simple react app that gives many errors regarding tree shaking disabled.
Why it disable tree shaking for react?
```
import ReactDOM from "react-dom";
import React from "react";
import { BrowserRouter, Routes, Route } from "react-router-dom";

export default function App() {
  return (
    <BrowserRouter>
      <Routes>
        <Route path="/" element={<Div>test</Div>} />

      </Routes>
    </BrowserRouter>
  );
}

ReactDOM.render(<App /> ,document.getElementById("root"));
```


@parcel/transformer-js: Unknown usage of CommonJS `exports` object. This causes 

tree shaking to be disabled.
  /Users//src/main/client/index.jsx:1:1

  > 1 | import ReactDOM from "react-dom";
  >   | ^
    2 | import { createRoot } from "react-dom/client";
    3 | 
  📝 Learn more: https://parceljs.org/features/scope-hoisting/#commonjs

Building jsx-dev-runtime.js...
Building index.js...
Building index.js...

@parcel/transformer-js: Conditional or non-top-level `require()` call. This 
causes the resolved module and all dependencies to be wrapped.

  /Users//src/main/client/node_modules/react/jsx-dev-runtime.js:6:20

    5 | } else {
  > 6 | ts = require('./cjs/react-jsx-dev-runtime.development.js');
  >   |      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    7 | }
    8 | 
  📝 Learn more: 
     https://parceljs.org/features/scope-hoisting/#avoid-conditional-require()

🚨 Build failed.

@parcel/core: Failed to resolve 'react-dom/client' from './index.jsx'

  /Users//src/main/client/index.jsx:2:28
    1 | import ReactDOM from "react-dom";
  > 2 | import { createRoot } from "react-dom/client";
  >   |                            ^^^^^^^^^^^^^^^^^^
    3 | 
    4 | import React from "react";

Building index.jsx...
Building index.js...
Building esmodule-helpers.js...
Building helpers.js...



@parcel/transformer-js: Unknown usage of CommonJS `exports` object. This causes 
tree shaking to be disabled.
  /Users//src/main/client/index.jsx:1:1
  > 1 | import ReactDOM from "react-dom";
  >   | ^
    2 | 
    3 | import React from "react";
  📝 Learn more: https://parceljs.org/features/scope-hoisting/#commonjs

Building react.development.js...
Building react-dom.development.js...
Building react-jsx-dev-runtime.development.js...
Building runtime.js...

@parcel/transformer-js: Conditional or non-top-level `require()` call. This 

causes the resolved module and all dependencies to be wrapped.

  /Users//src/main/client/node_modules/react/cjs/react.development.js:18:5
    17 | 
  > 18 | var _assign = require('object-assign');
  >    |     ^^^^^^^
    19 | var checkPropTypes = require('prop-types/checkPropTypes');
    20 | 
  📝 Learn more: 
     https://parceljs.org/features/scope-hoisting/#avoid-conditional-require()

@parcel/transformer-js: Conditional or non-top-level `require()` call. This 
causes the resolved module and all dependencies to be wrapped.


  📝 Learn more: 
     https://parceljs.org/features/scope-hoisting/#avoid-conditional-require()
  /Users//src/main/client/node_modules/react/cjs/react.development.js:19:5
    18 | var _assign = require('object-assign');
  > 19 | var checkPropTypes = require('prop-types/checkPropTypes');
  >    |     ^^^^^^^^^^^^^^
    20 | 
    21 | var ReactVersion = '16.14.0';

@parcel/transformer-js: Conditional or non-top-level `require()` call. This 
causes the resolved module and all dependencies to be wrapped.
  /Users//src/main/client/node_modules/react/cjs/react-jsx-dev-runtime.development.js:16:5
    15 | 
  > 16 | var React = require('react');
  >    |     ^^^^^
    17 | 


  📝 Learn more: 
     https://parceljs.org/features/scope-hoisting/#avoid-conditional-require()

    18 | // ATTENTION

@parcel/transformer-js: Conditional or non-top-level `require()` call. This 
causes the resolved module and all dependencies to be wrapped.
  /Users//src/main/client/node_modules/react-refresh/runtime.js:6:20
    5 | } else {
  > 6 | ts = require('./cjs/react-refresh-runtime.development.js');
  >   |      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    7 | }
    8 | 


  📝 Learn more: 
     https://parceljs.org/features/scope-hoisting/#avoid-conditional-require()

Building react-refresh-runtime.development.js...
Building checkPropTypes.js...

@parcel/transformer-js: Non-static access of an `import` or `require`. This 
causes tree shaking to be disabled for the resolved module.
  /Users//src/main/client/node_modules/react/cjs/react.development.js:254:11
    253 |   // Used by renderers to avoid bundling object-assign twice in UMD bu
  > 254 |   assign: _assign
  >     |           ^^^^^^^
    255 | };
    256 | 


  📝 Learn more: 
     https://parceljs.org/features/scope-hoisting/#dynamic-member-accesses
Building index.js...

@parcel/transformer-js: Conditional or non-top-level `require()` call. This 
causes the resolved module and all dependencies to be wrapped.

  /Users//src/main/client/node_modules/react-dom/cjs/react-dom.development.js:18:5
    17 | 
  > 18 | var React = require('react');
  >    |     ^^^^^
    19 | var _assign = require('object-assign');
    20 | var Scheduler = require('scheduler');

  📝 Learn more: 
     https://parceljs.org/features/scope-hoisting/#avoid-conditional-require()
Building index.js...
Building tracing.js...

@parcel/transformer-js: Non-static access of an `import` or `require`. This 


causes tree shaking to be disabled for the resolved module.
  /Users//src/main/client/node_modules/react/cjs/react.development.js:258:3
    257 | {
  > 258 |   _assign(ReactSharedInternals, {
  >     |   ^^^^^^^
    259 |     // These should not be included in production.
    260 |     ReactDebugCurrentFrame: ReactDebugCurrentFrame,
  📝 Learn more: 
     https://parceljs.org/features/scope-hoisting/#dynamic-member-accesses

@parcel/transformer-js: Non-static access of an `import` or `require`. This 

causes tree shaking to be disabled for the resolved module.
  /Users//src/main/client/node_modules/prop-types/checkPropTypes.js:61:96
    60 |           }
  > 61 | ull, ReactPropTypesSecret);
  >    |      ^^^^^^^^^^^^^^^^^^^^
    62 |         } catch (ex) {
    63 |           error = ex;

  📝 Learn more: 
     https://parceljs.org/features/scope-hoisting/#dynamic-member-accesses


@parcel/transformer-js: Conditional or non-top-level `require()` call. This 
causes the resolved module and all dependencies to be wrapped.
  /Users//src/main/client/node_modules/react-dom/cjs/react-dom.development.js:19:5
    18 | var React = require('react');
  > 19 | var _assign = require('object-assign');
  >    |     ^^^^^^^
    20 | var Scheduler = require('scheduler');
    21 | var checkPropTypes = require('prop-types/checkPropTypes');


  📝 Learn more: 
     https://parceljs.org/features/scope-hoisting/#avoid-conditional-require()
Building index.jsx...

@parcel/transformer-js: Non-static access of an `import` or `require`. This 


causes tree shaking to be disabled for the resolved module.
  /Users//src/main/client/node_modules/react/cjs/react.development.js:540:1
    539 | 
  > 540 | _assign(pureComponentPrototype, Component.prototype);
  📝 Learn more: 
     https://parceljs.org/features/scope-hoisting/#dynamic-member-accesses
  >     | ^^^^^^^
    541 | 
    542 | pureComponentPrototype.isPureReactComponent = true;
Building index.js...
Building tracing.js...

@parcel/transformer-js: Conditional or non-top-level `require()` call. This 

causes the resolved module and all dependencies to be wrapped.

  /Users//src/main/client/node_modules/react-dom/cjs/react-dom.development.js:20:5
    19 | var _assign = require('object-assign');
  > 20 | var Scheduler = require('scheduler');
  >    |     ^^^^^^^^^
    21 | var checkPropTypes = require('prop-types/checkPropTypes');
    22 | var tracing = require('scheduler/tracing');
  📝 Learn more: 
     https://parceljs.org/features/scope-hoisting/#avoid-conditional-require()

@parcel/transformer-js: Unknown usage of CommonJS `exports` object. This causes 


tree shaking to be disabled.
  📝 Learn more: https://parceljs.org/features/scope-hoisting/#commonjs

  /Users//src/main/client/index.jsx:1:1
  > 1 | import ReactDOM from "react-dom";
  >   | ^
    2 | import React from "react";
    3 | import { BrowserRouter, Routes, Route } from "react-router-dom";

@parcel/transformer-js: Non-static access of an `import` or `require`. This 


causes tree shaking to be disabled for the resolved module.
  /Users//src/main/client/node_modules/react/cjs/react.development.js:826:15
    825 | 
  > 826 | ps = _assign({}, element.props); // Reserved names are extracted
  >     |      ^^^^^^^
    827 | 
    828 | 
  📝 Learn more: 
     https://parceljs.org/features/scope-hoisting/#dynamic-member-accesses

@parcel/transformer-js: Non-static access of an `import` or `require`. This 
causes tree shaking to be disabled for the resolved module.
  /Users//src/main/client/node_modules/react/cjs/react.development.js:1714:7


    1713 |       setCurrentlyValidatingElement(element);
  > 1714 |      checkPropTypes(propTypes, element.props, 'prop', name, ReactDebu
  >      |      ^^^^^^^^^^^^^^
    1715 |       setCurrentlyValidatingElement(null);
    1716 |     } else if (type.PropTypes !== undefined && !propTypesMisspellWarn
  📝 Learn more: 
     https://parceljs.org/features/scope-hoisting/#dynamic-member-accesses


@parcel/transformer-js: Conditional or non-top-level `require()` call. This 


causes the resolved module and all dependencies to be wrapped.
  /Users//src/main/client/node_modules/react-dom/cjs/react-dom.development.js:21:5
    20 | var Scheduler = require('scheduler');
  > 21 | var checkPropTypes = require('prop-types/checkPropTypes');
  >    |     ^^^^^^^^^^^^^^
    22 | var tracing = require('scheduler/tracing');
    23 | 
  📝 Learn more: 
     https://parceljs.org/features/scope-hoisting/#avoid-conditional-require()
Building ReactPropTypesSecret.js...
Building scheduler.development.js...
Building scheduler-tracing.development.js...

@parcel/transformer-js: Conditional or non-top-level `require()` call. This 
causes the resolved module and all dependencies to be wrapped.


  /Users//src/main/client/node_modules/react-dom/cjs/react-dom.development.js:22:5
    21 | var checkPropTypes = require('prop-types/checkPropTypes');
  > 22 | var tracing = require('scheduler/tracing');
  >    |     ^^^^^^^
    23 | 
    24 | var ReactSharedInternals = React.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_W
  📝 Learn more: 
     https://parceljs.org/features/scope-hoisting/#avoid-conditional-require()

@parcel/transformer-js: Non-static access of an `import` or `require`. This 
causes tree shaking to be disabled for the resolved module.
  /Users//src/main/client/node_modules/react-dom/cjs/react-dom.development.js:103:6
    102 | 
  > 103 | if (!React) {
  >     |      ^^^^^
    104 |   {
    105 |     throw Error( "ReactDOM was loaded before React. Make sure you load


  📝 Learn more: 
     https://parceljs.org/features/scope-hoisting/#dynamic-member-accesses
Building index.js...
Building router.js...



  📝 Learn more: 
     https://parceljs.org/features/scope-hoisting/#dynamic-member-accesses
@parcel/transformer-js: Non-static access of an `import` or `require`. This 
causes tree shaking to be disabled for the resolved module.
  /Users//src/main/client/node_modules/react-dom/cjs/react-dom.development.js:1679:5
    1678 |   ReactControlledValuePropTypes.checkPropTypes = function (tagName, p
  > 1679 |     checkPropTypes(propTypes, props, 'prop', tagName, ReactDebugCurre
  >      |     ^^^^^^^^^^^^^^
    1680 |   };
    1681 | }

@parcel/transformer-js: Non-static access of an `import` or `require`. This 
causes tree shaking to be disabled for the resolved module.
  /Users//src/main/client/node_modules/react-dom/cjs/react-dom.development.js:1821:19
    1820 | 
  > 1821 |   var hostProps = _assign({}, props, {
  >      |                   ^^^^^^^
    1822 |     defaultChecked: undefined,
    1823 |     defaultValue: undefined,


  📝 Learn more: 
     https://parceljs.org/features/scope-hoisting/#dynamic-member-accesses
Bundling...

@parcel/transformer-js: Non-static access of an `import` or `require`. This 

causes tree shaking to be disabled for the resolved module.
  /Users//src/main/client/node_modules/react-dom/cjs/react-dom.development.js:2131:19
    2130 | function getHostProps$1(element, props) {
  > 2131 |   var hostProps = _assign({
  >      |                   ^^^^^^^
    2132 |     children: undefined
    2133 |   }, props);

  📝 Learn more: 
     https://parceljs.org/features/scope-hoisting/#dynamic-member-accesses

@parcel/transformer-js: Non-static access of an `import` or `require`. This 
causes tree shaking to be disabled for the resolved module.


  /Users//src/main/client/node_modules/react-dom/cjs/react-dom.development.js:2256:10
    2255 | function getHostProps$2(element, props) {
  📝 Learn more: 
     https://parceljs.org/features/scope-hoisting/#dynamic-member-accesses
  > 2256 |   return _assign({}, props, {
  >      |          ^^^^^^^
    2257 |     value: undefined
    2258 |   });

@parcel/transformer-js: Non-static access of an `import` or `require`. This 
causes tree shaking to be disabled for the resolved module.
  /Users//src/main/client/node_modules/react-dom/cjs/react-dom.development.js:2349:19
    2348 | 
  > 2349 |   var hostProps = _assign({}, props, {
  >      |                   ^^^^^^^
    2350 |     value: undefined,
    2351 |     defaultValue: undefined,


  📝 Learn more: 
     https://parceljs.org/features/scope-hoisting/#dynamic-member-accesses

@parcel/transformer-js: Non-static access of an `import` or `require`. This 


causes tree shaking to be disabled for the resolved module.
  /Users//src/main/client/node_modules/react-dom/cjs/react-dom.development.js:4713:23
    4712 | 
  > 4713 | var voidElementTags = _assign({
  📝 Learn more: 
     https://parceljs.org/features/scope-hoisting/#dynamic-member-accesses
  >      |                       ^^^^^^^
    4714 |   menuitem: true
    4715 | }, omittedCloseTags);



@parcel/transformer-js: Non-static access of an `import` or `require`. This 
causes tree shaking to be disabled for the resolved module.
  /Users//src/main/client/node_modules/react-dom/cjs/react-dom.development.js:7108:24
    7107 |   updatedAncestorInfo = function (oldInfo, tag) {
  > 7108 |     var ancestorInfo = _assign({}, oldInfo || emptyAncestorInfo);
  >      |                        ^^^^^^^
    7109 | 
    7110 |     var info = {
  📝 Learn more: 
     https://parceljs.org/features/scope-hoisting/#dynamic-member-accesses

@parcel/transformer-js: Non-static access of an `import` or `require`. This 
causes tree shaking to be disabled for the resolved module.

  /Users//src/main/client/node_modules/react-dom/cjs/react-dom.development.js:8423:1
    8422 | 
  > 8423 | _assign(SyntheticEvent.prototype, {
  >      | ^^^^^^^
    8424 |   preventDefault: function () {
    8425 |     this.defaultPrevented = true;

  📝 Learn more: 
     https://parceljs.org/features/scope-hoisting/#dynamic-member-accesses

@parcel/transformer-js: Non-static access of an `import` or `require`. This 
causes tree shaking to be disabled for the resolved module.


  📝 Learn more: 
     https://parceljs.org/features/scope-hoisting/#dynamic-member-accesses
  /Users//src/main/client/node_modules/react-dom/cjs/react-dom.development.js:8524:3
    8523 | 
  > 8524 |   _assign(prototype, Class.prototype);
  >      |   ^^^^^^^
    8525 | 
    8526 |   Class.prototype = prototype;

@parcel/transformer-js: Non-static access of an `import` or `require`. This 
causes tree shaking to be disabled for the resolved module.


  /Users//src/main/client/node_modules/react-dom/cjs/react-dom.development.js:8528:21
    8527 |   Class.prototype.constructor = Class;
  > 8528 |   Class.Interface = _assign({}, Super.Interface, Interface);
  >      |                     ^^^^^^^
    8529 |   Class.extend = Super.extend;
  📝 Learn more: 
     https://parceljs.org/features/scope-hoisting/#dynamic-member-accesses
    8530 |   addEventPoolingTo(Class);

@parcel/transformer-js: Non-static access of an `import` or `require`. This 
causes tree shaking to be disabled for the resolved module.
  /Users//src/main/client/node_modules/react-dom/cjs/react-dom.development.js:10741:7
    10740 |       var name = getComponentName(type) || 'Unknown';
  > 10741 |      checkPropTypes(contextTypes, context, 'context', name, getCurre
  >       |      ^^^^^^^^^^^^^^
    10742 |     } // Cache unmasked context so we can avoid recreating masked co
    10743 |     // Context is created before the class component is instantiated


  📝 Learn more: 
     https://parceljs.org/features/scope-hoisting/#dynamic-member-accesses

@parcel/transformer-js: Non-static access of an `import` or `require`. This 
causes tree shaking to be disabled for the resolved module.
  /Users//src/main/client/node_modules/react-dom/cjs/react-dom.development.js:10829:7
    10828 |       var name = getComponentName(type) || 'Unknown';
  > 10829 |      checkPropTypes(childContextTypes, childContext, 'child context'
  >       |      ^^^^^^^^^^^^^^
    10830 |       // somebody calls unstable_renderSubtreeIntoContainer() and we
    10831 |       // context from the parent component instance. The stack will 


  📝 Learn more: 
     https://parceljs.org/features/scope-hoisting/#dynamic-member-accesses


@parcel/transformer-js: Non-static access of an `import` or `require`. This 
causes tree shaking to be disabled for the resolved module.
  /Users//src/main/client/node_modules/react-dom/cjs/react-dom.development.js:10837:12
    10836 | 
  > 10837 |     return _assign({}, parentContext, {}, childContext);
  >       |            ^^^^^^^
    10838 |   }
    10839 | }

  📝 Learn more: 
     https://parceljs.org/features/scope-hoisting/#dynamic-member-accesses


@parcel/transformer-js: Non-static access of an `import` or `require`. This 

causes tree shaking to be disabled for the resolved module.
  /Users//src/main/client/node_modules/react-dom/cjs/react-dom.development.js:11846:17
    11845 |     // Resolve default props. Taken from ReactElement
  > 11846 |     var props = _assign({}, baseProps);
  >       |                 ^^^^^^^
    11847 | 
    11848 |     var defaultProps = Component.defaultProps;
  📝 Learn more: 
     https://parceljs.org/features/scope-hoisting/#dynamic-member-accesses

@parcel/transformer-js: Non-static access of an `import` or `require`. This 
causes tree shaking to be disabled for the resolved module.
  /Users//src/main/client/node_modules/react-dom/cjs/react-dom.development.js:12309:16
    12308 | 
  > 12309 |         return _assign({}, prevState, partialState);
  >       |                ^^^^^^^
    12310 |       }
    12311 | 


  📝 Learn more: 
     https://parceljs.org/features/scope-hoisting/#dynamic-member-accesses

@parcel/transformer-js: Non-static access of an `import` or `require`. This 

causes tree shaking to be disabled for the resolved module.
  /Users//src/main/client/node_modules/react-dom/cjs/react-dom.development.js:12610:89
    12609 | 

  > 12610 | te : _assign({}, prevState, partialState);
  >       |      ^^^^^^^
    12611 |   workInProgress.memoizedState = memoizedState; // Once the update q
    12612 |   // base state.
  📝 Learn more: 
     https://parceljs.org/features/scope-hoisting/#dynamic-member-accesses



@parcel/transformer-js: Non-static access of an `import` or `require`. This 
causes tree shaking to be disabled for the resolved module.
  📝 Learn more: 
     https://parceljs.org/features/scope-hoisting/#dynamic-member-accesses
  /Users//src/main/client/node_modules/react-dom/cjs/react-dom.development.js:16801:9
    16800 |       if (innerPropTypes) {
  > 16801 |         checkPropTypes(innerPropTypes, nextProps, // Resolved props
  >       |         ^^^^^^^^^^^^^^
    16802 |         'prop', getComponentName(Component), getCurrentFiberStackInD
    16803 |       }

@parcel/transformer-js: Non-static access of an `import` or `require`. This 


causes tree shaking to be disabled for the resolved module.
  /Users//src/main/client/node_modules/react-dom/cjs/react-dom.development.js:16870:9
    16869 |         // We could move it there, but we'd still need this for lazy
  > 16870 |         checkPropTypes(innerPropTypes, nextProps, // Resolved props
  >       |         ^^^^^^^^^^^^^^
    16871 |         'prop', getComponentName(type), getCurrentFiberStackInDev);
    16872 |       }
  📝 Learn more: 
     https://parceljs.org/features/scope-hoisting/#dynamic-member-accesses

@parcel/transformer-js: Non-static access of an `import` or `require`. This 
causes tree shaking to be disabled for the resolved module.
  /Users//src/main/client/node_modules/react-dom/cjs/react-dom.development.js:16889:7
    16888 |       // We could move it there, but we'd still need this for lazy c
  > 16889 |       checkPropTypes(_innerPropTypes, nextProps, // Resolved props
  >       |       ^^^^^^^^^^^^^^
    16890 |       'prop', getComponentName(_type), getCurrentFiberStackInDev);
    16891 |     }


  📝 Learn more: 
     https://parceljs.org/features/scope-hoisting/#dynamic-member-accesses



@parcel/transformer-js: Non-static access of an `import` or `require`. This 
causes tree shaking to be disabled for the resolved module.
  /Users//src/main/client/node_modules/react-dom/cjs/react-dom.development.js:16938:9
    16937 |       if (outerPropTypes) {
  > 16938 |      checkPropTypes(outerPropTypes, nextProps, // Resolved (SimpleMe
  >       |      ^^^^^^^^^^^^^^
    16939 |         'prop', getComponentName(outerMemoType), getCurrentFiberStac
    16940 |       } // Inner propTypes will be validated in the function compone
  📝 Learn more: 
     https://parceljs.org/features/scope-hoisting/#dynamic-member-accesses

@parcel/transformer-js: Non-static access of an `import` or `require`. This 
causes tree shaking to be disabled for the resolved module.
  /Users//src/main/client/node_modules/react-dom/cjs/react-dom.development.js:17015:9
    17014 |       if (innerPropTypes) {
  > 17015 |         checkPropTypes(innerPropTypes, nextProps, // Resolved props
  >       |         ^^^^^^^^^^^^^^
    17016 |         'prop', getComponentName(Component), getCurrentFiberStackInD

    17017 |       }

  📝 Learn more: 
     https://parceljs.org/features/scope-hoisting/#dynamic-member-accesses



  📝 Learn more: 
     https://parceljs.org/features/scope-hoisting/#dynamic-member-accesses
@parcel/transformer-js: Non-static access of an `import` or `require`. This 
causes tree shaking to be disabled for the resolved module.
  /Users//src/main/client/node_modules/react-dom/cjs/react-dom.development.js:17065:9
    17064 |       if (innerPropTypes) {
  > 17065 |         checkPropTypes(innerPropTypes, nextProps, // Resolved props
  >       |         ^^^^^^^^^^^^^^
    17066 |         'prop', getComponentName(Component), getCurrentFiberStackInD
    17067 |       }


@parcel/transformer-js: Non-static access of an `import` or `require`. This 
causes tree shaking to be disabled for the resolved module.
  /Users//src/main/client/node_modules/react-dom/cjs/react-dom.development.js:17379:15
    17378 |             if (outerPropTypes) {
  > 17379 |      checkPropTypes(outerPropTypes, resolvedProps, // Resolved for o
  >       |      ^^^^^^^^^^^^^^
    17380 |               'prop', getComponentName(Component), getCurrentFiberSt
    17381 |             }

  📝 Learn more: 
     https://parceljs.org/features/scope-hoisting/#dynamic-member-accesses

@parcel/transformer-js: Non-static access of an `import` or `require`. This 
causes tree shaking to be disabled for the resolved module.


  /Users//src/main/client/node_modules/react-dom/cjs/react-dom.development.js:18233:7
    18232 |     if (providerPropTypes) {
  > 18233 |      checkPropTypes(providerPropTypes, newProps, 'prop', 'Context.Pr
  >       |      ^^^^^^^^^^^^^^
    18234 |     }
  📝 Learn more: 
     https://parceljs.org/features/scope-hoisting/#dynamic-member-accesses
    18235 |   }

@parcel/transformer-js: Non-static access of an `import` or `require`. This 


causes tree shaking to be disabled for the resolved module.
  /Users//src/main/client/node_modules/react-dom/cjs/react-dom.development.js:18675:15
    18674 |             if (outerPropTypes) {
  > 18675 |      checkPropTypes(outerPropTypes, _resolvedProps3, // Resolved for
  📝 Learn more: 
     https://parceljs.org/features/scope-hoisting/#dynamic-member-accesses
  >       |      ^^^^^^^^^^^^^^
    18676 |               'prop', getComponentName(_type2), getCurrentFiberStack
    18677 |             }


@parcel/transformer-js: Non-static access of an `import` or `require`. This 

causes tree shaking to be disabled for the resolved module.
  📝 Learn more: 
     https://parceljs.org/features/scope-hoisting/#dynamic-member-accesses
  /Users//src/main/client/node_modules/react-dom/cjs/react-dom.development.js:24482:54
    24481 |     var key = path[idx];
  > 24482 | () : _assign({}, obj); // $FlowFixMe number or string is fine here
  >       |      ^^^^^^^
    24483 | 
    24484 |     updated[key] = copyWithSetImpl(obj[key], path, idx + 1, value);

@parcel/transformer-js: Non-static access of an `import` or `require`. This 
causes tree shaking to be disabled for the resolved module.
  /Users//src/main/client/node_modules/react-dom/cjs/react-dom.development.js:24512:29
    24511 | 
  > 24512 |       fiber.memoizedProps = _assign({}, fiber.memoizedProps);
  >       |                             ^^^^^^^
    24513 |       scheduleWork(fiber, Sync);
    24514 |     }


  📝 Learn more: 
     https://parceljs.org/features/scope-hoisting/#dynamic-member-accesses

@parcel/transformer-js: Non-static access of an `import` or `require`. This 
causes tree shaking to be disabled for the resolved module.
  /Users//src/main/client/node_modules/react-dom/cjs/react-dom.development.js:24540:26
    24539 |   var ReactCurrentDispatcher = ReactSharedInternals.ReactCurrentDisp
  > 24540 |   return injectInternals(_assign({}, devToolsConfig, {
  >       |                          ^^^^^^^
    24541 |     overrideHookState: overrideHookState,
    24542 |     overrideProps: overrideProps,


  📝 Learn more: 
     https://parceljs.org/features/scope-hoisting/#dynamic-member-accesses

Building runtime-6a0402d3ec776393.js...
Building runtime-402a68cf08055057.js...
Building runtime.js...
Building index.js...
Building react-refresh-runtime.development.js...
Building browser.js...
Packaging index.html...
Packaging index.[hash].js...
Optimizing index.html...
Optimizing index.[hash].js...
✨ Built in 34.85s% EXECUTING [50s]
Building index.jsx...
Bundling...

@parcel/transformer-js: Unknown usage of CommonJS `exports` object. This causes 
tree shaking to be disabled.
  /Users//src/main/client/index.jsx:1:1
  > 1 | import ReactDOM from "react-dom";
  >   | ^
    2 | import React from "react";
    3 | import { BrowserRouter, Routes, Route } from "react-router-dom";


  📝 Learn more: https://parceljs.org/features/scope-hoisting/#commonjs
